pkg/plugins/generic/generic_plugin.go

```diff
@@ -323,13 +346,13 @@ func (p *GenericPlugin) syncDesiredKernelArgs(kargs []string) (bool, error) {
 	return needReboot, nil
 }
 
-func (p *GenericPlugin) needDrainNode(desired sriovnetworkv1.Interfaces, current sriovnetworkv1.InterfaceExts) (needDrain bool) {
+func (p *GenericPlugin) needDrainNode(desired sriovnetworkv1.SriovNetworkNodeStateSpec, current sriovnetworkv1.SriovNetworkNodeStateStatus) (needDrain bool) {
 	log.Log.V(2).Info("generic plugin needDrainNode()", "current", current, "desired", desired)
 
 	needDrain = false
-	for _, ifaceStatus := range current {
+	for _, ifaceStatus := range current.Interfaces {
```

Comment:
now that this function check both interfaces and bridges can you move this to another function like `NeedToUpdateVfs`?

Reply:
done